Motherlode Mine - Sack overlay and region check

ypperlig opened the pull request:
The following changes have been done:

* Added red color to the "Pay-dirt in sack" overlay when the number of pay-dirt in sack exceeds or is equal to the max allowed.
* Added "Deposits left" to sack overlay. This shows the number of "inventories" left before the sack will be full. This is colored red when deposits left are 0.
* Removed "You are mining" / "You are not mining" as this could not be shown accurately. This showed "You are not mining" when animation stopped, although the character would continue mining straight after.
* Added isInMLM (region check) to most functions

Screenshots:
![Overlay](https://user-images.githubusercontent.com/4660715/38391415-711b4816-3924-11e8-85f9-aafb944397f4.png)
![OverlayRed](https://user-images.githubusercontent.com/4660715/38391316-31372a58-3924-11e8-9431-13171549fcf8.png)

SoyChai commented:
Great additions ypperlig, just be prepared to split this into 2 pull requests since Adam isn't a fan of combining many changes into one merge :)

psikoi commented:
The "You are mining" is still useful, couldn't you have a timer, and if the animation stopped for more than x milliseconds it would count as not mining? This would prevent the plugin mistakenly showing "You are not mining" for the small animation pauses, would it not?

deathbeam commented:
But then "You are mining" would kind of lose purpose because it will not be accurate anymore, and for inacurrate idle alerts you have idle notifier.

psikoi commented:
If the animation pauses for 700ms lets say, then it would only be inaccurate when the player stopped mining, only for 700ms, that sounds like a reasonable downside to having the overlay

deathbeam commented:
No, I mean, the purpose of the You are fishing, you are woodcutting, you are mining overlays is to be accurate, more accurate than idle notifier. So current you are mining should either stay as it is now or be removed. I would probably just don't change it.

ypperlig commented:
The general consensus on Discord when I brought it up was that it should be removed as it could not be shown accurately like Fishing/Woodcutting.

If it shouldn't be changed (stay as is) then I think a toggle should be introduced.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java:
```diff
@@ -56,19 +69,27 @@
+	public int CUR_SACK_SIZE;
+	public int MAX_SACK_SIZE;
+
+	private static final Set<Integer> MOTHERLODE_MAP_REGIONS = Sets.newHashSet(14935, 14936, 14937, 15191, 15192, 15193);
```
`ImmutableSet.of`

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOverlay.java:
```diff
@@ -90,17 +71,6 @@ public Dimension render(Graphics2D graphics)
 
 		panelComponent.getLines().clear();
 
-		if (MINING_ANIMATION_IDS.contains(client.getLocalPlayer().getAnimation()))
```
I think this is worth trying to keep. Maybe add a timeout if there is a small window where you still mining with no animation.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java:
```diff
@@ -114,6 +141,17 @@ public MotherlodeSession getSession()
+	{
+		if (isInMLM())
+		{
+			CUR_SACK_SIZE = client.getSetting(Varbits.SACK_NUMBER);
```
Non-constants should not be named with all caps. Also there is a varbit changed event maybe that would be better for this.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java:
```diff
@@ -205,18 +273,93 @@ public void onGameObjectChanged(GameObjectChanged event)
+		int regionY = localWorld.getY() >> 6;
+		int regionID = regionX << 8 | regionY;
+
+		return MOTHERLODE_MAP_REGIONS.contains(regionID);
```
There is `client.getMapRegions()`. Maybe it is worth putting a on region change event somewhere. We do have the gamestate change event which will go between region and logged in when the region changes.

Adam- replied:
Now that I think about this, just checking this when the gamestate changes to LOGGED_IN would work fine.

ypperlig commented:
Updated PR with the following changes:
* Moved region check to onRegionChange and onGameStateChanged. Introduced plugin.inMLM rather than isInMLM() for region check.
* Switched to client.getMapRegions() to fetch current regions. Region check loops through all regions and verifies they exist in MOTHERLODE_MAP_REGIONS.
* Added 'You are mining' / 'You are currently NOT mining' back. No changes have been done to this. Introduced a toggle that disables this.
